### DISCUSSION TITLE
QInput + mask + numbers only + max characters length + numeric keypad

### DISCUSSION THREAD

**Discussion by nulele** [quasarframework/quasar#10082] [CLI - Capacitor / Cordova modes] [answered]
Hello,

for my cordova mode app I'd like to have an input field for numbers only, with 3 max characters length and the numeric keypad.
I cannot find the right combination.

      <q-input
        v-model:number="height"
        type="number"
        suffix="cm"
        mask="###"
        pattern="[0-9]*"
        maxlength="3"
        label="Height"
      />

This allows me to digit more than 3 characters.
If I remove the _mask_ or _type_ attribute the keypad is alpha.
_pattern_ attribute seems useless.
_maxlength_ attribute doesn't work either

Curiosity: this combination allows the "e" letter... maybe a bug?

Any idea? Thanks

**Comment by MilosPaunovic** [comment]
Hello @nulele,

As per [official docs](https://quasar.dev/vue-components/input#mask) warning:

`Mask is only available if the type is one of ‘text’ (default), ‘search’, ‘url’, ‘tel’, or ‘password’.` 

Your type is number, so it won't work. If you remove that line, works just fine.

**Comment by nulele** [comment]
Hi @MilosPaunovic ,

thanks, you're right, but if I change the _type_ attribute to 'text' the keypad is alpha.

**Reply by MilosPaunovic** [reply]
Could you 'cheat' and use `type="tel"`?

**Reply by nulele** [reply]
I was trying it right now... the keypad shows extra characters but with the _mask_ attribute only numbers are accepted.
Not the best solution but the closest. Thanks!

**Comment by hawkeye64** [comment, marked as answer]
It's a Vue bug: https://github.com/vuejs/vue-next/issues/3813
Fixed in v3.2.0-beta.2: https://github.com/vuejs/vue-next/blob/master/CHANGELOG.md
